src/dependencies/useStoreDependency.ts

```diff
@@ -67,15 +76,13 @@ function useStoreDependency<Props>(
     dispatchToken,
     dependencyValue,
     dependencyMap,
+    props,
+    prevProps,
   ]);
-
-  React.useMemo(() => {
-    const newValue = calculate(dependency, props);
-    if (newValue !== dependencyValue) {
-      setDependencyValue(newValue);
-    }
-  }, propsArray);
-
+  const newValue = calculate(dependency, props);
+  if (newValue !== dependencyValue) {
+    setDependencyValue(newValue);
```

Comment:
could this be `setDependencyValue(calculate(dependency, props));` because of https://reactjs.org/docs/hooks-reference.html#bailing-out-of-a-state-update

Reply:
unfortunately not. it causes an infinite loop

Reply:
ok interesting. hooks are confusing!